packages/starlight/utils/navigation.ts

```diff
@@ -10,11 +10,17 @@ import { ensureLeadingAndTrailingSlashes, ensureTrailingSlash } from './path';
 
 const DirKey = Symbol('DirKey');
 
+type Badge = {
+	text: string;
+	variant: 'blue' | 'pink' | 'green' | 'yellow' | 'purple';
+}
```

Comment:
This type could be inferred directly from the schema to avoid maintaining it manually maybe? If you end up extracting the schema to its own file, this file could also exports a `Badge` type, something like this quick mockup _(no idea how to create a GitHub suggestion for a potential new file)_:

```ts
import { z } from 'astro/zod';

const badgeSchema = () =>
  z.object({
    variant: z
      .enum(["blue", "pink", "green", "yellow", "purple"])
      .default("blue"),
    text: z.string(),
  });

export const BadgeConfigSchema = () =>
  z.union([z.string(), badgeSchema()]).optional();

export type Badge = z.output<ReturnType<typeof badgeSchema>>;
```

Reply:
Great points. Creating a new file will definitely be the best way do it.